model/jpa/src/main/java/org/keycloak/models/workflow/RestartWorkflowTask.java

```diff
@@ -0,0 +1,26 @@
+package org.keycloak.models.workflow;
+
+import org.jboss.logging.Logger;
+import org.keycloak.models.KeycloakSession;
+
+class RestartWorkflowTask extends RunWorkflowTask {
+
+    private static final Logger log = Logger.getLogger(RunWorkflowTask.class);
+
+    private final DefaultWorkflowExecutionContext context;
+
+    RestartWorkflowTask(DefaultWorkflowExecutionContext context) {
+        super(context, false);
+        this.context = context;
+    }
+
+    @Override
+    public void run(KeycloakSession session) {
+        Workflow workflow = context.getWorkflow();
+        String resourceId = context.getResourceId();
+        String executionId = context.getExecutionId();
+        context.resetState();
```

Comment:
is it intentional? The class is named "restart", but you're invoking "resetState".